pkg/cmd/operator/gatherbase.go

```diff
@@ -107,13 +103,24 @@ func (o *GatherBaseOptions) Complete() error {
 
 	o.discoveryClient = cacheddiscovery.NewMemCacheClient(o.kubeClient.Discovery())
 
+	ignoreDestDirExists := false
 	if len(o.DestDir) == 0 {
 		o.DestDir = fmt.Sprintf("%s-%s", o.GathererName, utilrand.String(12))
-		err := os.Mkdir(o.DestDir, 0770)
-		if err != nil {
-			return fmt.Errorf("can't create destination directory %q: %w", o.DestDir, err)
-		}
+	} else {
+		// We have already made sure that the dir doesn't exist or is empty in validation.
```

Comment:
Wouldn't the logic be simpler if you moved DestDir creation before verification?

Reply:
/hold

Reply:
I don't think it should create dest dir when there are invalid arguments. The point of verification as a first step is to make sure the inputs are correct before taking any action.

Reply:
Fair enough. In that case I'd personally move the destDir creation out of the Validate func though.

Reply:
it's inside "Complete()" function...

Reply:
ah, nvm, didn't notice the lines skipped by github